File: services/language-service/src/main/java/dev/animedia/languageservice/application/usecase/SearchLanguageUseCase.java
package dev.animedia.languageservice.application.usecase;

import dev.animedia.languageservice.application.dto.LanguageDto;
import dev.animedia.languageservice.domain.model.Page;
import dev.animedia.languageservice.application.dto.SearchLanguageDto;

public interface SearchLanguageUseCase {
    Page<LanguageDto> search(SearchLanguageDto searchDto);
}

File: services/social-service/src/core/Application/Language/UseCase/SearchLanguageUseCase.php
<?php

namespace Core\Application\Language\UseCase;

use Core\Application\Language\DTO\LanguageDto;
use Core\Domain\Shared\Pagination\Entity\Page;
use Core\Domain\Shared\Pagination\Entity\Pageable;

interface SearchLanguageUseCase
{
    /**
     * @return Page<LanguageDto>
     */
    public function execute(?string $name, ?string $languageIsoCode, ?bool $isActive, Pageable $pageable): Page;
}
